pkg/controller/core/workload_controller.go

```diff
@@ -136,27 +136,14 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
-		// because we need to react to API cluster queue events, the list of checks from a cache can lead to race conditions
-		queueAdmissionChecks, err := r.getClusterQueueAdmissionChecks(ctx, cqName)
-		if err == nil {
-			newChecks, shouldUpdate := syncAdmissionCheckConditions(wl.Status.AdmissionChecks, queueAdmissionChecks)
-			if shouldUpdate {
-				log.V(3).Info("The workload needs admission checks updates", "clusterQueue", klog.KRef("", cqName), "queueChecks", queueAdmissionChecks)
-				wl.Status.AdmissionChecks = newChecks
-				err := r.client.Status().Update(ctx, &wl)
-				return ctrl.Result{}, client.IgnoreNotFound(err)
-			}
+		if updated, err := r.reconcileSyncAdmissionChecks(ctx, &wl, cqName); updated {
```

Comment:
`updated || err != nil`
Just in case `reconcileSyncAdmissionChecks` implementation changes in the future in a way that would result in an error even if updated is false.

Reply:
done